### PR TITLE
LPD-19577 - Import/Export Object Definitions in a root model Integration tests

### PR DESCRIPTION
**Pull request by G-Sa**
Related to [LPD-19577](https://liferay.atlassian.net/browse/LPD-19577)

### CODE REVIEW

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/portlet/action/test/BaseExportImportTestCase.java [hunk 1]
```diff
@@ -178,13 +133,111 @@ private ThemeDisplay _getThemeDisplay() throws Exception {
...
+		}
+
+		mockLiferayPortletActionRequest.addParameter(
+			"redirect", RandomTestUtil.randomString());
+
+		mockLiferayPortletActionRequest.setAttribute(
+			JavaConstants.JAVAX_PORTLET_CONFIG, null);
+
+		mockLiferayPortletActionRequest.setAttribute(
+			WebKeys.THEME_DISPLAY, _getThemeDisplay());
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/portlet/action/test/BaseExportImportTestCase.java:207]: @G-Sa you can remove the empty lines

File: modules/apps/object/object-test/src/testIntegration/resources/com/liferay/object/web/internal/portlet/action/test/dependencies/test-invalid-bound-object-definitions.json [hunk 2]
```diff
@@ -0,0 +1,680 @@
...
+		"objectViews": [
+		],
+		"panelCategoryKey": "",
+		"parameterRequired": false,
+		"pluralLabel": {
+			"en_US": "roots"
+		},
+		"portlet": true,
+		"restContextPath": "/o/c/roots",
+		"rootObjectDefinitionExternalReferenceCode": "root",
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/resources/com/liferay/object/web/internal/portlet/action/test/dependencies/test-invalid-bound-object-definitions.json:236]: @G-Sa should we follow the same pattern from the other json files? something like "`TESTROOTBJECTDEFINITION`"

File: modules/apps/object/object-test/src/testIntegration/resources/com/liferay/object/web/internal/portlet/action/test/dependencies/test-invalid-bound-object-definitions.json [hunk 3]
```diff
@@ -0,0 +1,680 @@
...
+		"objectViews": [
+		],
+		"panelCategoryKey": "",
+		"parameterRequired": false,
+		"pluralLabel": {
+			"en_US": "branchs"
+		},
+		"portlet": false,
+		"restContextPath": "/o/c/roots/branches",
+		"rootObjectDefinitionExternalReferenceCode": "root",
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/resources/com/liferay/object/web/internal/portlet/action/test/dependencies/test-invalid-bound-object-definitions.json:478]: @G-Sa same here

File: modules/apps/object/object-test/src/testIntegration/resources/com/liferay/object/web/internal/portlet/action/test/dependencies/test-invalid-bound-object-definitions.json [hunk 4]
```diff
@@ -0,0 +1,680 @@
...
+		"objectViews": [
+		],
+		"panelCategoryKey": "",
+		"parameterRequired": false,
+		"pluralLabel": {
+			"en_US": "leaves"
+		},
+		"portlet": false,
+		"restContextPath": "/o/c/roots/leafs",
+		"rootObjectDefinitionExternalReferenceCode": "root",
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/resources/com/liferay/object/web/internal/portlet/action/test/dependencies/test-invalid-bound-object-definitions.json:670]: @G-Sa same here

File: modules/apps/object/object-test/src/testIntegration/resources/com/liferay/object/web/internal/portlet/action/test/dependencies/test-invalid-bound-object-definitions.json [hunk 5]
```diff
@@ -0,0 +1,680 @@
...
+			}
+		},
+		"active": false,
+		"defaultLanguageId": "en_US",
+		"enableCategorization": true,
+		"enableComments": false,
+		"enableLocalization": false,
+		"enableObjectEntryDraft": false,
+		"enableObjectEntryHistory": false,
+		"externalReferenceCode": "root",
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/resources/com/liferay/object/web/internal/portlet/action/test/dependencies/test-invalid-bound-object-definitions.json:28]: @G-Sa same here

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by carolmariaabb** [comment]
@G-Sa maybe you need to add the feature flag for these tests.

**Comment by G-Sa** [comment]
ci:test:sf

**Comment by G-Sa** [comment]
ci:test:relevant

**Comment by G-Sa** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:stop

**Comment by G-Sa** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
Related failure:

1. https://testray.liferay.com/home/-/testray/case_results/2558401000

image

**Comment by G-Sa** [comment]
I could not reproduce that behavior locally, both tests that failed there passed even in an environment where the feature flag is not active (being turned on only by the tests)

**Pull request closed without merging**